src/model/writer.js

```diff
@@ -795,17 +860,22 @@ export default class Writer {
 		if ( !newRange ) {
 			// If `newRange` is not given, treat this as synchronizing existing marker.
 			// Create `MarkerOperation` with `oldRange` set to `null`, so reverse operation will remove the marker.
-			addMarkerOperation( this, name, null, currentRange );
+			addMarkerOperation( this, markerName, null, currentRange );
 		} else {
 			// Just change marker range.
-			addMarkerOperation( this, name, currentRange, newRange );
+			addMarkerOperation( this, markerName, currentRange, newRange );
```

Comment:
Maybe `applyMarkerOperation`? It's strange that you add a marker. Add to what? It also sounds like it is about adding markers what is not true.

Reply:
I agree. WIll fix it.